adminSiteClient/GdocsIndexPage.tsx

```diff
@@ -1,155 +1,281 @@
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome/index.js"
-import { AdminAppContext } from "./AdminAppContext.js"
-import { OwidGdocTag, OwidGdocInterface } from "@ourworldindata/utils"
+import {
+    OwidGdocTag,
+    OwidGdocInterface,
+    OwidGdocType,
+} from "@ourworldindata/utils"
 import { Route, RouteComponentProps } from "react-router-dom"
 import { Link } from "./Link.js"
 import { GdocsAdd } from "./GdocsAdd.js"
-import { Observer } from "mobx-react"
-import { useGdocsStore } from "./GdocsStore.js"
-import { runInAction } from "mobx"
+import { observer } from "mobx-react"
+import { GdocsStoreContext } from "./GdocsStore.js"
+import { computed, observable } from "mobx"
+import { BAKED_BASE_URL } from "../settings/clientSettings.js"
+import fuzzysort from "fuzzysort"
+
+const iconGdocTypeMap = {
+    [OwidGdocType.Fragment]: <FontAwesomeIcon icon={faPuzzlePiece} />,
+    [OwidGdocType.Article]: <FontAwesomeIcon icon={faNewspaper} />,
+    [OwidGdocType.TopicPage]: <FontAwesomeIcon icon={faLightbulb} />,
+}
+
+interface Searchable {
+    gdoc: OwidGdocInterface
+    term?: Fuzzysort.Prepared
+}
+
+@observer
+class GdocsIndexPageSearch extends React.Component<{
+    filters: Record<OwidGdocType, boolean>
+    search: { value: string }
+}> {
+    handleCheckboxChange = (type: OwidGdocType) => {
+        this.props.filters[type] = !this.props.filters[type]
+    }
+
+    render() {
+        const owidGdocTypes: OwidGdocType[] = [
+            OwidGdocType.Fragment,
+            OwidGdocType.Article,
+            OwidGdocType.TopicPage,
+        ]
+
+        return (
+            <div className="d-flex flex-grow-1">
+                <SearchField
+                    placeholder="author, category, title"
+                    className="gdoc-index__search-bar"
+                    value={this.props.search.value}
+                    onValue={(value: string) =>
+                        (this.props.search.value = value)
+                    }
+                    autofocus
+                />
+                <div>
+                    {owidGdocTypes.map((type) => {
+                        const checked = this.props.filters[type]
+                        return (
+                            <label
+                                key={type}
+                                className={cx("gdoc-index-filter-checkbox", {
+                                    checked,
+                                })}
+                            >
+                                <input
+                                    type="checkbox"
+                                    id={`shouldShow${type}`}
+                                    checked={checked}
+                                    onChange={() =>
+                                        this.handleCheckboxChange(type)
+                                    }
+                                />
+                                {iconGdocTypeMap[type]}
+                            </label>
```

Comment:
I find the filtering behaviour a bit surprising. Traditionally, all checkboxes are unticked and mean "no filters applied, show all". Then ticking one activates filtering on this particular facet, which feels more straightforward than deselecting what you don't want to see.

I like that icons are used in the filters and repeated in the search results. This helps with visual filtering without actually using the search functions. It would be better though if both the checkboxes and their labels were visible, especially given that the meaning of these icons is not universal.